File: Olivia/tree/Symmetric tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root==null ){//|| help(root.left,root.right)){
            return true;
        }
        return help(root.left,root.right);
    }
    public boolean help(TreeNode left, TreeNode right){
        if(left==null &&right==null){
            return true;
        }
        if((left==null &&right!=null)||(left!=null &&right==null)){
            return false;
        }
        if(left.val!=right.val){
            return false;
        }
        return help(left.left,right.right) && help(left.right,right.left);
    }
}

File: Sen/Symmetric_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} root
    # @return {boolean}

### recursive solution 1: 
    def isSymmetric(self, root):
	    if not root:
	    	return True
	    else:
	    	return self.isMirror(root.left, root.right)

	def isMirror(self, left, right):
		if left == right == None: 
			return True
		if left is None or right is None:
			return False
		if left.val == right.val:
			outPair = self.isMirror(left.left, right.right)
			inPair = self.isMirror(left.right, right.left)
			return outPair and inPair
		else:
			return False
    ### simplied version
	# def isMirror(self, left, right):
	# 	if node1 and node2:
	# 		return node1.val == node2.val and self.isMirror(left.left, right.right) and self.isMirror(left.right, right.left)
	# 	else:
	# 		return left == right